packages/scaffold/src/modal/w3m-modal/index.ts

```diff
@@ -59,6 +58,21 @@ export class W3mModal extends LitElement {
     }
   }
 
+  private initializeTheming() {
+    initializeTheming(ThemeController.state.themeVariables)
+    if (ThemeController.state.themeMode) {
+      ThemeController.setThemeMode(ThemeController.state.themeMode)
+    } else if (typeof window !== 'undefined' && window.matchMedia) {
```

Comment:
Can we abstract all of this into ui helper that returns just light or dark value to be set.